packages/report-server/src/reportBuilder/output/functions/matrix/matrix.ts

```diff
@@ -0,0 +1,58 @@
+/**
+ * Tupaia
+ * Copyright (c) 2017 - 2020 Beyond Essential Systems Pty Ltd
+ */
+
+import { Matrix, Row } from '../../../types';
+import { MatrixBuilder } from './matrixBuilder';
+import { MatrixColumnParams, MatrixParams } from './types';
+
+const matrix = (rows: Row[], params: MatrixParams): Matrix => {
+  const matrixBuilder = new MatrixBuilder(rows, params);
+  matrixBuilder.buildBaseColumns().buildBaseRows();
```

Comment:
It might be nicer if the matrix builder itself were responsible for calling the sub-functions, so that the interface could just be:
```suggestion
  return matrixBuilder.build();
```

with build being: 
```
build() {
  this.buildBaseColumns();
  this.buildBaseRows();
  return this.matrixData;
}
```

Also, just a small naming thing, but why `buildBaseColumns()`/`buildBaseRows()` instead of just `buildColumns()`/`buildRows()` ?

Reply:
That's a good question @rohan-bes ! I was thinking we will need more functions here to decorate rows and columns, same as those from `web-config-server`, such as adding total summary at the end of each row, but your question inspires me that `transform` can also handle this case.

`transform` is awesome!